aes/src/x86/vaes256/aes128.rs

```diff
@@ -0,0 +1,533 @@
+use crate::x86::{arch::*, RoundKeys, Simd256RoundKeys};
+use crate::{Block, Block32};
+use cipher::inout::InOut;
+use core::{arch::asm, mem::MaybeUninit};
+
+#[inline(always)]
+pub(crate) unsafe fn parallelize_keys(keys: &RoundKeys<11>) -> Simd256RoundKeys<11> {
+    let mut v256: [MaybeUninit<__m256i>; 11] = MaybeUninit::uninit().assume_init();
+    asm! {
+        "vbroadcastf128 ymm0 , [{keys} +  0 * 16]",
+        "vbroadcastf128 ymm1 , [{keys} +  1 * 16]",
+        "vbroadcastf128 ymm2 , [{keys} +  2 * 16]",
+        "vbroadcastf128 ymm3 , [{keys} +  3 * 16]",
+        "vbroadcastf128 ymm4 , [{keys} +  4 * 16]",
+        "vbroadcastf128 ymm5 , [{keys} +  5 * 16]",
+        "vbroadcastf128 ymm6 , [{keys} +  6 * 16]",
+        "vbroadcastf128 ymm7 , [{keys} +  7 * 16]",
+        "vbroadcastf128 ymm8 , [{keys} +  8 * 16]",
+        "vbroadcastf128 ymm9 , [{keys} +  9 * 16]",
+        "vbroadcastf128 ymm10, [{keys} + 10 * 16]",
+
+        "vmovdqu [{optr} +  0 * 32], ymm0 ",
+        "vmovdqu [{optr} +  1 * 32], ymm1 ",
+        "vmovdqu [{optr} +  2 * 32], ymm2 ",
+        "vmovdqu [{optr} +  3 * 32], ymm3 ",
+        "vmovdqu [{optr} +  4 * 32], ymm4 ",
+        "vmovdqu [{optr} +  5 * 32], ymm5 ",
+        "vmovdqu [{optr} +  6 * 32], ymm6 ",
+        "vmovdqu [{optr} +  7 * 32], ymm7 ",
+        "vmovdqu [{optr} +  8 * 32], ymm8 ",
+        "vmovdqu [{optr} +  9 * 32], ymm9 ",
+        "vmovdqu [{optr} + 10 * 32], ymm10",
+
+        keys = in(reg) keys.as_ptr(),
+        optr = in(reg) v256.as_mut_ptr().cast::<__m256i>(),
+
+        clobber_abi("C"),
+        options(nostack, preserves_flags),
+    };
+    core::mem::transmute(v256)
+}
+
+#[inline]
+pub(crate) unsafe fn encrypt32(
+    keys: &RoundKeys<11>,
+    simd_256_keys: &Simd256RoundKeys<11>,
+    blocks: InOut<'_, '_, Block32>,
+) {
+    let (iptr, optr) = blocks.into_raw();
+    asm! {
+        // load plain-data
+        "vmovdqu ymm1 , [{iptr} +  0 * 32]",
+        "vmovdqu ymm2 , [{iptr} +  1 * 32]",
+        "vmovdqu ymm3 , [{iptr} +  2 * 32]",
+        "vmovdqu ymm4 , [{iptr} +  3 * 32]",
+        "vmovdqu ymm5 , [{iptr} +  4 * 32]",
+        "vmovdqu ymm6 , [{iptr} +  5 * 32]",
+        "vmovdqu ymm7 , [{iptr} +  6 * 32]",
+        "vmovdqu ymm8 , [{iptr} +  7 * 32]",
+        "vmovdqu ymm9 , [{iptr} +  8 * 32]",
+        "vmovdqu ymm10, [{iptr} +  9 * 32]",
+        "vmovdqu ymm11, [{iptr} + 10 * 32]",
+        "vmovdqu ymm12, [{iptr} + 11 * 32]",
+        "vmovdqu ymm13, [{iptr} + 12 * 32]",
+        "vmovdqu ymm14, [{iptr} + 13 * 32]",
+        "vmovdqu ymm15, [{iptr} + 14 * 32]",
+        // aes-128 round 0 encrypt
+        "vmovdqu ymm0 , [{simd_256_keys} +  0 * 32]",
```

Comment:
Ideally, we would keep the round keys in registers (and adjust number of blocks processed in parallel accordingly). We probably can do it in this case since `__m256i` is available on stable.